umh.docs.umh.app/content/en/docs/datacontracts/_index.md

```diff
@@ -0,0 +1,213 @@
+---
+title: "Data Contracts / API"
+menuTitle: "Data Contracts / API"
+description: "This page describes how messages flow in the UMH, which message goes where, how it has to be formatted and how you can create your own structures."
+weight: 1725
+---
+
+## What are Data Contracts
+
+Data Contracts are formal agreements that define the structure, format, and
+rules for data exchanged between components within a Unified Namespace (UNS)
+architecture. They specify metadata, data models, and service levels to ensure
+that all systems interact consistently and reliably. By establishing clear
+guidelines for data exchange, Data Contracts facilitate seamless integration
+and prevent disruptions caused by changes in data formats or structures.
+
+In short, they define where a message is going, what structure it must follow,
+how it gets there, and what happens when it arrives, based on defined rules and
+services.
+
+### Example Historian
+
+To give you a simple example, just think about the `_historian' schema. Perhaps
+without realizing it, you have already used the Historian Data Contract by using
+this schema. The Historian Data Contract governs the flow of data from your
+machines to the database by enforcing rules and providing services:
+
+- **Protocol Converter**: You must use a protocol converter to send your data
```

Comment:
these are the components right? but we introduce the components only at a later stage.

maybe here it makes more sense to explain what one can do with it? sop to explain the results:

To give you a simple example, just think about the `_historian' schema. Perhaps
without realizing it, you have already used the Historian Data Contract by using
this schema.

Whenever you send a message in via MQTT into this data contract, you will know that it will land up in the tag or tag_string tables in TimescaleDB and that they will be bridged to Kafka as well. You could also send it directly into Kafka, and you would know that it gets bridged to MQTT as well.

Reply:
If there are any issues, e.g., schema issues, you would know that you could look them up in the Management Console.

Reply:
in order for the data contract to work, the incoming messages into _historian from MQTT and Kafka need to follow a topic hierarchy and payload schema, which we as UMH have defined. If you don't like it, you can also define your own custom data contracts.